packages/components/src/web-preview/index.js

```diff
@@ -0,0 +1,90 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { Component, createRef } from '@wordpress/element';
+import { noop } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Spinner from '../spinner';
+
+/**
+ * WebPreview component to display an iframe of another page.
+ */
+class WebPreview extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			isLoading: true,
+		};
+
+		this.iframeRef = createRef();
+		this.setLoaded = this.setLoaded.bind( this );
+	}
+
+	componentDidMount() {
+		this.iframeRef.current.addEventListener( 'load', this.setLoaded );
+	}
+
+	setLoaded() {
+		this.setState( { isLoading: false } );
+		this.props.onLoad();
+	}
+
+	render() {
+		const { className, loading, src, title } = this.props;
+		const { isLoading } = this.state;
+
+		const classes = classnames( 'woocommerce-web-preview', className, {
+			'is-loading': isLoading,
+		} );
+
+		return (
+			<div className={ classes }>
+				{ isLoading && loading }
+				<div className="woocommerce-web-preview__iframe-wrapper">
+					<iframe
+						ref={ this.iframeRef }
+						title={ title }
+						src={ src }
+					/>
+				</div>
+			</div>
+		);
+	}
+}
+
+WebPreview.propTypes = {
+	/**
+	 * Additional class name to style the component.
+	 */
+	className: PropTypes.string,
+	/**
+	 * Content shown when iframe is still loading.
+	 */
+	loading: PropTypes.node,
```

Comment:
This prop name made me do a double take - perhaps it is just too close to `isLoading`. What do you think about `loadingContent` or `loadingComponent`?

Reply:
Agreed.  I originally named it `loadingContent` but that threw an error (reserved name if I remember correctly).  Will update this before merging.